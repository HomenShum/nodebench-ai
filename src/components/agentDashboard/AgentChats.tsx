import React, { useMemo } from "react";
import { useQuery } from "convex/react";
import { api } from "../../../convex/_generated/api";
import { Id } from "../../../convex/_generated/dataModel";

export function AgentChats() {
  const threads = useQuery(api.chatThreads.listThreadsForUser as any, { limit: 50 }) as Array<{
    threadId: Id<"documents">;
    title: string;
    lastModified: number;
    messageCount: number;
    lastMessage?: { role: "user" | "assistant"; text: string; createdAt: number };
  }> | undefined;

  const rows = useMemo(() => (threads ?? []), [threads]);

  const openThread = (docId: Id<"documents">) => {
    try {
      // Hint the app shell to open this document
      window.dispatchEvent(new CustomEvent("ai:openMultipleDocuments", { detail: { documentIds: [docId] } }));
      // Fallback: navigate to Documents hub
      const base = "#documents";
      if (window.location.hash !== base) window.location.hash = base;
    } catch {}
  };

  return (
    <div className="h-full overflow-auto bg-[var(--bg-secondary)]">
      <div className="max-w-7xl mx-auto p-4 sm:p-6">
        <div className="mb-4 flex items-center justify-between">
          <div>
            <h2 className="text-base font-semibold text-[var(--text-primary)]">Chat History</h2>
            <p className="text-xs text-[var(--text-secondary)]">Threads saved from the AI Chat Panel</p>
          </div>
        </div>

        <div className="overflow-auto border border-[var(--border-color)] rounded-md bg-white">
          <table className="min-w-full text-xs">
            <thead className="bg-[var(--bg-primary)] text-[var(--text-secondary)]">
              <tr>
                <th className="text-left px-3 py-2 border-b">Title</th>
                <th className="text-left px-3 py-2 border-b">Last Message</th>
                <th className="text-left px-3 py-2 border-b">Messages</th>
                <th className="text-left px-3 py-2 border-b">Updated</th>
                <th className="text-left px-3 py-2 border-b">Actions</th>
              </tr>
            </thead>
            <tbody>
              {rows.map((t) => {
                const updated = new Date(t.lastModified || 0).toLocaleString();
                const last = t.lastMessage ? `${t.lastMessage.role === 'user' ? 'You' : 'Assistant'}: ${t.lastMessage.text}` : 'â€”';
                return (
                  <tr key={String(t.threadId)} className="odd:bg-white even:bg-[var(--bg-secondary)]">
                    <td className="px-3 py-2 border-b">
                      <button className="underline hover:no-underline" onClick={() => openThread(t.threadId)}>{t.title}</button>
                    </td>
                    <td className="px-3 py-2 border-b max-w-[520px] truncate" title={t.lastMessage?.text || ''}>{last}</td>
                    <td className="px-3 py-2 border-b">{t.messageCount}</td>
                    <td className="px-3 py-2 border-b whitespace-nowrap">{updated}</td>
                    <td className="px-3 py-2 border-b">
                      <button className="text-xs px-2 py-1 rounded border border-[var(--border-color)] hover:bg-[var(--bg-hover)]" onClick={() => openThread(t.threadId)}>Open</button>
                    </td>
                  </tr>
                );
              })}
              {rows.length === 0 && (
                <tr>
                  <td className="px-3 py-6 text-center text-[var(--text-secondary)]" colSpan={5}>No chat threads yet. Enable auto-save in the AI Chat Panel to record conversations.</td>
                </tr>
              )}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}

