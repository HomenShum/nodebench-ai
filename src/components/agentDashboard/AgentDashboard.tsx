import { useMemo, useState, useEffect } from "react";
import { useQuery, useMutation, useConvex } from "convex/react";
import { api } from "../../../convex/_generated/api";
import { Id } from "../../../convex/_generated/dataModel";
import { AgentTimeline } from "@/components/agentDashboard/AgentTimeline";
import { AgentTasks } from "@/components/agentDashboard/AgentTasks";
import { AgentChats } from "@/components/agentDashboard/AgentChats";
import "@/styles/agentDashboard.css";
import { AgentWindowProvider } from "./AgentWindowContext";

export type AgentDashboardTab = "timeline" | "tasks" | "chat";

export function AgentDashboard() {
  const [tab, setTab] = useState<AgentDashboardTab>("tasks");
  const timelines = useQuery(api.agentTimelines.listForUser, {}) as Array<{
    timelineId: Id<"agentTimelines">; documentId: Id<"documents">; title: string; updatedAt: number;
  }> | undefined;
  const sorted = useMemo(() => (timelines ?? []).slice().sort((a, b) => b.updatedAt - a.updatedAt), [timelines]);
  const [selectedId, setSelectedId] = useState<Id<"agentTimelines"> | null>(null);
  const selectedTimelineId = selectedId ?? (sorted[0]?.timelineId ?? null);

  // Deep-link: allow hash #calendar/agents?timeline=<id> and custom event 'agents:openTimeline'
  useEffect(() => {
    function trySelectFromHash() {
      try {
        const h = window.location.hash || "";
        if (!h.startsWith("#calendar/agents")) return;
        const qIndex = h.indexOf("?");
        if (qIndex === -1) return;
        const search = new URLSearchParams(h.slice(qIndex + 1));
        const tl = search.get("timeline");
        if (tl) {
          setSelectedId(tl as unknown as Id<"agentTimelines">);
          setTab("timeline");
        }
      } catch {}
    }
    const onHash = () => trySelectFromHash();
    window.addEventListener("hashchange", onHash);
    trySelectFromHash();

    const onOpen = (e: Event) => {
      const ev = e as CustomEvent<{ timelineId?: string }>;
      const tl = ev.detail?.timelineId;
      if (tl) {
        setSelectedId(tl as unknown as Id<"agentTimelines">);
        setTab("timeline");
        try {
          const base = "#calendar/agents";
          const hash = `${base}?timeline=${encodeURIComponent(tl)}`;
          if (window.location.hash !== hash) window.location.hash = hash;
        } catch {}
      }
    };
    window.addEventListener("agents:openTimeline", onOpen as EventListener);
    return () => {
      window.removeEventListener("hashchange", onHash);
      window.removeEventListener("agents:openTimeline", onOpen as EventListener);
    };
  }, [setSelectedId, setTab]);



  const convex = useConvex();
  const createDoc = useMutation(api.documents.create);
  const createTimeline = useMutation(api.agentTimelines.createForDocument);
  const updateTask = useMutation(api.agentTimelines.updateTaskMetrics);
  const [isRunningTest, setIsRunningTest] = useState(false);

  // Global actions from popovers/buttons (depends on updateTask)
  useEffect(() => {
    const onAction = async (e: Event) => {
      const ev = e as CustomEvent<{ action: string; task: any }>;
      const { action, task } = ev.detail || ({} as any);
      if (!task || !task._id) return;
      try {
        if (action === 'pause') await updateTask({ taskId: task._id, status: 'paused' } as any);
        if (action === 'resume') await updateTask({ taskId: task._id, status: 'running', startedAtMs: Date.now() } as any);
        if (action === 'rerun') await updateTask({ taskId: task._id, status: 'running', progress: 0, startedAtMs: Date.now(), elapsedMs: 0 } as any);
      } catch (err) {
        console.error('Task action failed', err);
      }
    };
    window.addEventListener('agents:taskAction', onAction as EventListener);
    return () => {
      window.removeEventListener('agents:taskAction', onAction as EventListener);
    };
  }, [updateTask]);

  const handleCreateTimeline = async () => {
    try {
      const docId = await createDoc({ title: "Agents Hub Timeline", parentId: undefined, content: [] as any });
      const tlId = await createTimeline({ documentId: docId as Id<"documents">, name: "Agents Hub" });
      setSelectedId(tlId as Id<"agentTimelines">);
      // Hash deep-link for agents view remains
      try { window.location.hash = "#calendar/agents"; } catch {}
    } catch (err) {
      console.error(err);
      alert((err as any)?.message ?? "Failed to create timeline");
    }
  };

  const handleRunVisualLLMTest = async () => {
    if (isRunningTest) return;

    try {
      setIsRunningTest(true);

      // Create a new timeline for the test
      const docId = await createDoc({
        title: "Visual LLM Validation Test",
        parentId: undefined,
        content: [] as any
      });
      const tlId = await createTimeline({
        documentId: docId as Id<"documents">,
        name: "Visual LLM Test"
      });

      // Switch to the new timeline and timeline tab
      setSelectedId(tlId as Id<"agentTimelines">);
      setTab("timeline");

      // Run the visual LLM validation workflow
      // Note: This calls an internal action, so we need to use the convex client
      await convex.action(api.agents.orchestrate.run, {
        documentId: docId as Id<"documents">,
        name: "Visual LLM Validation Test",
        taskSpec: {
          goal: "Visual LLM validation workflow: search for test images, validate and filter them, analyze with GPT-5-mini and Gemini 2.5 Flash using real vision APIs, compare results, and recommend best model",
          type: "orchestrate",
          topic: "Visual LLM Model Validation for VR Avatar Quality Assessment (GPT-5-mini vs Gemini 2.5 Flash)",
          graph: {
            nodes: [
              {
                id: "image_search",
                kind: "search",
                label: "Search VR Avatar Test Images",
                prompt: "VR avatars virtual reality characters full body 3D models hands feet eyes clothing",
              },
              {
                id: "image_validation",
                kind: "custom",
                label: "Validate Image URLs",
                prompt: "{{channel:image_search.last}}",
              },
              {
                id: "image_filtering",
                kind: "custom",
                label: "Filter Valid Images",
                prompt: "{{channel:image_validation.last}}",
              },
              {
                id: "vision_analysis",
                kind: "custom",
                label: "Parallel Vision Analysis (GPT-5-mini + Gemini 2.5 Flash)",
                prompt: "{{channel:image_filtering.last}}",
              },
              {
                id: "statistical_analysis",
                kind: "custom",
                label: "Statistical Analysis & Aggregation",
                prompt: "{{channel:vision_analysis.last}}",
              },
              {
                id: "visualization",
                kind: "custom",
                label: "Generate Plotly Visualizations",
                prompt: "{{channel:statistical_analysis.last}}",
              },
              {
                id: "model_comparison",
                kind: "structured",
                label: "Model Performance Comparison",
                prompt: "Based on statistical analysis {{channel:statistical_analysis.last}} and visualizations {{channel:visualization.last}}, compare GPT-5-mini vs Gemini 2.5 Flash.",
              },
              {
                id: "prompt_optimization",
                kind: "answer",
                label: "Enhanced Prompt Generation",
                prompt: "Based on model comparison {{channel:model_comparison.last}}, generate enhanced prompts to improve GPT-5-mini and Gemini 2.5 Flash performance.",
              },
              {
                id: "eval_quality",
                kind: "eval",
                label: "Quality Check & Follow-up",
                prompt: "Evaluate the completeness of the analysis. Check if all steps completed successfully.",
              },
            ],
            edges: [
              { from: "image_search", to: "image_validation" },
              { from: "image_validation", to: "image_filtering" },
              { from: "image_filtering", to: "vision_analysis" },
              { from: "vision_analysis", to: "statistical_analysis" },
              { from: "statistical_analysis", to: "visualization" },
              { from: "visualization", to: "model_comparison" },
              { from: "model_comparison", to: "prompt_optimization" },
              { from: "prompt_optimization", to: "eval_quality" },
            ],
          },
        },
      });

      // Success notification
      alert("Visual LLM validation test started! Check the Timeline tab for progress.");
    } catch (err) {
      console.error("Failed to run Visual LLM test:", err);
      alert((err as any)?.message ?? "Failed to run Visual LLM test. Make sure API keys are configured.");
    } finally {
      setIsRunningTest(false);
    }
  };

  return (
    <div className="agent-dashboard h-full w-full flex flex-col">
      {/* Header */}
      <div className="bg-[var(--bg-primary)] border-b border-[var(--border-color)] px-4 sm:px-6 py-3 flex items-center justify-between sticky top-0 z-10">
        <div className="flex items-center gap-3">
          <div className="w-8 h-8 rounded-lg bg-gradient-to-br from-[var(--accent-primary)] to-indigo-500 text-white flex items-center justify-center shadow">
            <span className="text-sm font-semibold">A</span>
          </div>
          <div>
            <div className="text-sm font-semibold text-[var(--text-primary)]">Multi-Agent Research</div>
            <div className="text-xs text-[var(--text-secondary)]">Timeline & Tasks</div>
          </div>
          {/* Timeline selector */}
          <div className="ml-3 inline-flex items-center gap-2">
            <select
              className="text-xs px-2 py-1 rounded-md border border-[var(--border-color)] bg-white"
              value={selectedTimelineId ?? ""}
              onChange={(e) => setSelectedId(e.target.value as unknown as Id<"agentTimelines">)}
            >
              {sorted.map((t) => (
                <option key={String(t.timelineId)} value={String(t.timelineId)}>{t.title}</option>
              ))}
            </select>
            <button
              className="px-2 py-1 text-xs rounded-md border bg-[var(--bg-primary)] border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
              onClick={handleCreateTimeline}
            >
              New Timeline
            </button>
            <button
              className={`px-3 py-1 text-xs rounded-md border font-medium transition-colors ${
                isRunningTest
                  ? "bg-gray-100 text-gray-400 border-gray-300 cursor-not-allowed"
                  : "bg-gradient-to-r from-purple-500 to-indigo-500 text-white border-transparent hover:from-purple-600 hover:to-indigo-600 shadow-sm"
              }`}
              onClick={handleRunVisualLLMTest}
              disabled={isRunningTest}
              title="Run Visual LLM validation test with GPT-5-mini and Gemini 2.5 Flash"
            >
              {isRunningTest ? (
                <>
                  <span className="inline-block animate-spin mr-1">⏳</span>
                  Running Test...
                </>
              ) : (
                <>
                  🧪 Run Visual LLM Test
                </>
              )}
            </button>
          </div>
        </div>
        <div className="inline-flex items-center gap-1 p-1 rounded-lg bg-[var(--bg-tertiary)] border border-[var(--border-color)]">
          <button
            className={`px-3 py-1.5 text-xs rounded-md border transition-colors ${tab === "tasks" ? "bg-white text-[var(--text-primary)] border-[var(--border-color)] shadow-sm" : "text-[var(--text-secondary)] border-transparent hover:bg-white/50"}`}
            onClick={() => setTab("tasks")}
          >
            Tasks
          </button>
          <button
            className={`px-3 py-1.5 text-xs rounded-md border transition-colors ${tab === "timeline" ? "bg-white text-[var(--text-primary)] border-[var(--border-color)] shadow-sm" : "text-[var(--text-secondary)] border-transparent hover:bg-white/50"}`}
            onClick={() => setTab("timeline")}
          >
            Timeline
          </button>
          <button
            className={`px-3 py-1.5 text-xs rounded-md border transition-colors ${tab === "chat" ? "bg-white text-[var(--text-primary)] border-[var(--border-color)] shadow-sm" : "text-[var(--text-secondary)] border-transparent hover:bg-white/50"}`}
            onClick={() => setTab("chat")}
          >
            Chat
          </button>
        </div>
      </div>

      {/* Body */}
      <AgentWindowProvider>
        <div className="flex-1 min-h-0">
          {selectedTimelineId ? (
            <>
              {/* Timeline Tab - Keep mounted, hide with CSS to prevent editor reappending */}
              <div style={{ display: tab === "timeline" ? "block" : "none" }} className="h-full">
                <AgentTimeline
                  timelineId={selectedTimelineId}
                  documentId={sorted.find(t => t.timelineId === selectedTimelineId)?.documentId as Id<'documents'> | undefined}
                />
              </div>

              {/* Chat Tab - Keep mounted, hide with CSS */}
              <div style={{ display: tab === "chat" ? "block" : "none" }} className="h-full">
                <AgentChats />
              </div>

              {/* Tasks Tab - Keep mounted, hide with CSS */}
              <div style={{ display: tab === "tasks" ? "block" : "none" }} className="h-full">
                <AgentTasks
                  timelineId={selectedTimelineId}
                  onViewTimeline={() => setTab("timeline")}
                />
              </div>
            </>
          ) : (
            tab === "chat" ? (
              <AgentChats />
            ) : (
              <div className="p-6 text-sm text-[var(--text-secondary)]">No timelines yet. Create one to get started.</div>
            )
          )}
        </div>
      </AgentWindowProvider>


    </div>
  );
}

