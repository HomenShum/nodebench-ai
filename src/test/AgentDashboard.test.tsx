// @vitest-environment jsdom
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

vi.mock('convex/react', async () => {
  const actual: any = await vi.importActual('convex/react');
  return {
    ...actual,
    useQuery: (_fn: any, args: any) => {
      // Heuristic: when querying a specific timeline, args has a timelineId
      if (args && typeof args === 'object' && 'timelineId' in args) {
        return {
          baseStartMs: Date.now() - 10000,
          tasks: [
            { _id: 't1', name: 'Plan', status: 'running', durationMs: 60000, startOffsetMs: 0 },
            { _id: 't2', name: 'Collect', status: 'pending', durationMs: 120000, startOffsetMs: 30000 },
          ],
          links: [],
        } as any;
      }
      // Otherwise it's the listForUser
      return [
        { timelineId: 'tl1' as any, title: 'Agents Hub', updatedAt: Date.now() },
        { timelineId: 'tl2' as any, title: 'Project X', updatedAt: Date.now() - 1000 },
      ];
    },
    useMutation: () => async () => undefined,
    useConvex: () => ({ query: vi.fn(), mutation: vi.fn() }),
  };
});

import { AgentDashboard } from '@/components/agentDashboard/AgentDashboard';

describe('AgentDashboard', () => {

  it('renders and switches tabs', () => {
    const { asFragment } = render(<AgentDashboard />);
    // Initial timeline tab
    expect(screen.getByText('Timeline & Tasks')).toBeTruthy();

    // Switch to Tasks tab
    fireEvent.click(screen.getByText('Tasks'));
    expect(screen.getByText('Agent Tasks')).toBeTruthy();
    // Basic sanity checks instead of brittle snapshots
    expect(screen.getAllByText('Open Full View').length).toBeGreaterThan(0);
    expect(screen.getAllByText(/Plan|Collect/).length).toBeGreaterThan(0);
  });
});

