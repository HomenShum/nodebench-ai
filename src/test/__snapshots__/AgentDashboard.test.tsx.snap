// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentDashboard > renders and switches tabs 1`] = `
<DocumentFragment>
  <div
    class="agent-dashboard h-full w-full flex flex-col"
  >
    <div
      class="bg-[var(--bg-primary)] border-b border-[var(--border-color)] px-4 sm:px-6 py-3 flex items-center justify-between sticky top-0 z-10"
    >
      <div
        class="flex items-center gap-3"
      >
        <div
          class="w-8 h-8 rounded-lg bg-gradient-to-br from-[var(--accent-primary)] to-indigo-500 text-white flex items-center justify-center shadow"
        >
          <span
            class="text-sm font-semibold"
          >
            A
          </span>
        </div>
        <div>
          <div
            class="text-sm font-semibold text-[var(--text-primary)]"
          >
            Multi-Agent Research
          </div>
          <div
            class="text-xs text-[var(--text-secondary)]"
          >
            Timeline & Tasks
          </div>
        </div>
        <div
          class="ml-3 inline-flex items-center gap-2"
        >
          <select
            class="text-xs px-2 py-1 rounded-md border border-[var(--border-color)] bg-white"
          >
            <option
              value="tl1"
            >
              Agents Hub
            </option>
            <option
              value="tl2"
            >
              Project X
            </option>
          </select>
          <button
            class="px-2 py-1 text-xs rounded-md border bg-[var(--bg-primary)] border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
          >
            New Timeline
          </button>
        </div>
      </div>
      <div
        class="inline-flex items-center gap-1 p-1 rounded-lg bg-[var(--bg-tertiary)] border border-[var(--border-color)]"
      >
        <button
          class="px-3 py-1.5 text-xs rounded-md border transition-colors bg-white text-[var(--text-primary)] border-[var(--border-color)] shadow-sm"
        >
          Timeline
        </button>
        <button
          class="px-3 py-1.5 text-xs rounded-md border transition-colors text-[var(--text-secondary)] border-transparent hover:bg-white/50"
        >
          Tasks
        </button>
      </div>
    </div>
    <div
      class="flex-1 min-h-0"
    >
      <div
        class="h-full grid"
        style="grid-template-rows: auto 1fr;"
      >
        <div
          class="bg-[var(--bg-primary)] border-b border-[var(--border-color)] px-4 py-2 flex items-center gap-2"
        >
          <span
            class="text-xs text-[var(--text-secondary)]"
          >
            Status:
          </span>
          <span
            class="px-2 py-0.5 text-[10px] rounded-full bg-emerald-50 border border-emerald-200 text-emerald-700 uppercase tracking-wide"
          >
            Active
          </span>
        </div>
        <div
          class="grid"
          style="grid-template-columns: 300px 1fr;"
        >
          <aside
            class="border-r border-[var(--border-color)] bg-white overflow-auto"
          >
            <div
              class="sticky top-0 z-10 h-12 border-b border-[var(--border-color)] flex items-center px-3 text-xs text-[var(--text-secondary)] uppercase tracking-wide bg-white"
            >
              Agent Hierarchy
            </div>
            <div
              class="divide-y divide-[var(--border-color)]"
            >
              <div
                class="flex items-center gap-2 px-3 h-12 hover:bg-[var(--bg-secondary)] cursor-default"
              >
                <div
                  class="w-6 h-6 rounded-md bg-[var(--bg-secondary)] border border-[var(--border-color)] flex items-center justify-center text-sm"
                >
                  ðŸ§©
                </div>
                <div
                  class="flex-1 min-w-0"
                >
                  <div
                    class="text-sm text-[var(--text-primary)] truncate"
                  >
                    agent
                  </div>
                  <div
                    class="text-[10px] text-[var(--text-muted)]"
                  >
                    2 task(s)
                  </div>
                </div>
                <span
                  class="w-2 h-2 rounded-full bg-emerald-500"
                />
              </div>
            </div>
          </aside>
          <div
            class="relative bg-white overflow-auto"
          >
            <div
              class="sticky top-0 z-10 bg-white border-b border-[var(--border-color)]"
            >
              <div
                class="h-10 flex items-center text-[10px] uppercase tracking-wide text-[var(--text-muted)] px-3"
              >
                Wall-clock
              </div>
              <div
                class="h-10 flex"
                data-grid="true"
              >
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  1
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  2
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  3
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  4
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  5
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  6
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  7
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  8
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  9
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  10
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  11
                </div>
                <div
                  class="flex-1 min-w-[80px] border-r border-dashed border-[var(--border-color)] text-[10px] flex items-center justify-center text-[var(--text-secondary)]"
                >
                  12
                </div>
              </div>
            </div>
            <div
              class="absolute top-[80px] bottom-0 w-px bg-[var(--accent-primary)]/70 pointer-events-none"
              style="left: 0px;"
            />
            <div
              class="relative"
            >
              <div
                class="h-12 border-b border-dashed border-[var(--border-color)] relative"
              >
                <div
                  class="absolute top-2 h-7 rounded-md bg-[var(--accent-primary)]/10 border border-[var(--accent-primary)]/30 px-2 text-xs flex items-center text-[var(--text-primary)] shadow-sm hover:shadow cursor-pointer"
                  style="left: 0px; width: 60px;"
                  title="Plan"
                >
                  Plan
                </div>
                <div
                  class="absolute top-2 h-7 rounded-md bg-[var(--accent-primary)]/10 border border-[var(--accent-primary)]/30 px-2 text-xs flex items-center text-[var(--text-primary)] shadow-sm hover:shadow cursor-pointer"
                  style="left: 30px; width: 120px;"
                  title="Collect"
                >
                  Collect
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AgentDashboard > renders and switches tabs 2`] = `
<DocumentFragment>
  <div
    class="agent-dashboard h-full w-full flex flex-col"
  >
    <div
      class="bg-[var(--bg-primary)] border-b border-[var(--border-color)] px-4 sm:px-6 py-3 flex items-center justify-between sticky top-0 z-10"
    >
      <div
        class="flex items-center gap-3"
      >
        <div
          class="w-8 h-8 rounded-lg bg-gradient-to-br from-[var(--accent-primary)] to-indigo-500 text-white flex items-center justify-center shadow"
        >
          <span
            class="text-sm font-semibold"
          >
            A
          </span>
        </div>
        <div>
          <div
            class="text-sm font-semibold text-[var(--text-primary)]"
          >
            Multi-Agent Research
          </div>
          <div
            class="text-xs text-[var(--text-secondary)]"
          >
            Timeline & Tasks
          </div>
        </div>
        <div
          class="ml-3 inline-flex items-center gap-2"
        >
          <select
            class="text-xs px-2 py-1 rounded-md border border-[var(--border-color)] bg-white"
          >
            <option
              value="tl1"
            >
              Agents Hub
            </option>
            <option
              value="tl2"
            >
              Project X
            </option>
          </select>
          <button
            class="px-2 py-1 text-xs rounded-md border bg-[var(--bg-primary)] border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
          >
            New Timeline
          </button>
        </div>
      </div>
      <div
        class="inline-flex items-center gap-1 p-1 rounded-lg bg-[var(--bg-tertiary)] border border-[var(--border-color)]"
      >
        <button
          class="px-3 py-1.5 text-xs rounded-md border transition-colors text-[var(--text-secondary)] border-transparent hover:bg-white/50"
        >
          Timeline
        </button>
        <button
          class="px-3 py-1.5 text-xs rounded-md border transition-colors bg-white text-[var(--text-primary)] border-[var(--border-color)] shadow-sm"
        >
          Tasks
        </button>
      </div>
    </div>
    <div
      class="flex-1 min-h-0"
    >
      <div
        class="h-full overflow-auto bg-[var(--bg-secondary)]"
      >
        <div
          class="max-w-7xl mx-auto p-4 sm:p-6"
        >
          <div
            class="mb-4 flex items-center justify-between"
          >
            <div>
              <h2
                class="text-base font-semibold text-[var(--text-primary)]"
              >
                Agent Tasks
              </h2>
              <p
                class="text-xs text-[var(--text-secondary)]"
              >
                Compact cards with hover previews
              </p>
            </div>
            <button
              class="px-3 py-1.5 text-xs rounded-md bg-[var(--accent-primary)] text-white hover:opacity-90"
            >
              + New Task
            </button>
          </div>
          <div
            class="grid gap-4 sm:gap-5 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
          >
            <div
              class="relative bg-white border border-[var(--border-color)] rounded-xl overflow-hidden hover:shadow transition cursor-default group"
            >
              <div
                class="hidden group-hover:flex absolute inset-0 z-10 bg-white/95 backdrop-blur px-4 py-3 flex-col"
              >
                <div
                  class="text-sm font-semibold"
                >
                  Plan
                </div>
                <div
                  class="text-xs text-[var(--text-secondary)]"
                >
                  Final output
                </div>
                <div
                  class="mt-2 p-2 bg-[var(--bg-secondary)] border border-[var(--border-color)] rounded text-xs leading-5 whitespace-pre-wrap"
                >
                  No output yet.
                </div>
                <div
                  class="mt-auto pt-3 flex gap-2"
                >
                  <button
                    class="px-2.5 py-1 text-xs rounded-md border border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
                  >
                    Open Full View
                  </button>
                  <button
                    class="px-2.5 py-1 text-xs rounded-md border border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
                  >
                    Accept
                  </button>
                  <button
                    class="px-2.5 py-1 text-xs rounded-md border border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
                  >
                    Reject
                  </button>
                </div>
              </div>
              <div
                class="p-3 border-b border-[var(--border-color)] bg-white"
              >
                <div
                  class="text-sm font-medium flex items-center gap-2"
                >
                  <span
                    class="text-lg"
                  >
                    ðŸ§©
                  </span>
                  <span
                    class="truncate"
                  >
                    Plan
                  </span>
                </div>
                <div
                  class="mt-1 text-xs text-[var(--text-secondary)] flex items-center justify-between"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <span
                      class="w-2 h-2 rounded-full bg-emerald-500"
                    />
                    <span
                      class="capitalize"
                    >
                      running
                    </span>
                  </div>
                  <span
                    class="text-[var(--text-muted)]"
                  >
                    ~60s
                  </span>
                </div>
              </div>
              <div
                class="p-3 bg-[var(--bg-tertiary)] border-t border-[var(--border-color)]"
              >
                <div
                  class="text-[10px] font-semibold uppercase tracking-wide text-[var(--text-secondary)]"
                >
                  Output
                </div>
                <div
                  class="mt-1 text-xs line-clamp-2 whitespace-pre-wrap text-[var(--text-primary)]"
                >
                  No output yet.
                </div>
              </div>
            </div>
            <div
              class="relative bg-white border border-[var(--border-color)] rounded-xl overflow-hidden hover:shadow transition cursor-default group"
            >
              <div
                class="hidden group-hover:flex absolute inset-0 z-10 bg-white/95 backdrop-blur px-4 py-3 flex-col"
              >
                <div
                  class="text-sm font-semibold"
                >
                  Collect
                </div>
                <div
                  class="text-xs text-[var(--text-secondary)]"
                >
                  Final output
                </div>
                <div
                  class="mt-2 p-2 bg-[var(--bg-secondary)] border border-[var(--border-color)] rounded text-xs leading-5 whitespace-pre-wrap"
                >
                  No output yet.
                </div>
                <div
                  class="mt-auto pt-3 flex gap-2"
                >
                  <button
                    class="px-2.5 py-1 text-xs rounded-md border border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
                  >
                    Open Full View
                  </button>
                  <button
                    class="px-2.5 py-1 text-xs rounded-md border border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
                  >
                    Accept
                  </button>
                  <button
                    class="px-2.5 py-1 text-xs rounded-md border border-[var(--border-color)] hover:bg-[var(--bg-hover)]"
                  >
                    Reject
                  </button>
                </div>
              </div>
              <div
                class="p-3 border-b border-[var(--border-color)] bg-white"
              >
                <div
                  class="text-sm font-medium flex items-center gap-2"
                >
                  <span
                    class="text-lg"
                  >
                    ðŸ§©
                  </span>
                  <span
                    class="truncate"
                  >
                    Collect
                  </span>
                </div>
                <div
                  class="mt-1 text-xs text-[var(--text-secondary)] flex items-center justify-between"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <span
                      class="w-2 h-2 rounded-full bg-amber-500"
                    />
                    <span
                      class="capitalize"
                    >
                      pending
                    </span>
                  </div>
                  <span
                    class="text-[var(--text-muted)]"
                  >
                    ~120s
                  </span>
                </div>
              </div>
              <div
                class="p-3 bg-[var(--bg-tertiary)] border-t border-[var(--border-color)]"
              >
                <div
                  class="text-[10px] font-semibold uppercase tracking-wide text-[var(--text-secondary)]"
                >
                  Output
                </div>
                <div
                  class="mt-1 text-xs line-clamp-2 whitespace-pre-wrap text-[var(--text-primary)]"
                >
                  No output yet.
                </div>
              </div>
            </div>
            <button
              class="min-h-[220px] border-2 border-dashed border-indigo-200 rounded-xl grid place-items-center bg-indigo-50/40 hover:bg-indigo-50 text-indigo-600"
            >
              <div
                class="text-center"
              >
                <div
                  class="w-12 h-12 rounded-xl border border-indigo-200 grid place-items-center bg-white mx-auto mb-2 text-lg"
                >
                  +
                </div>
                <div
                  class="text-sm font-semibold"
                >
                  New Task
                </div>
                <div
                  class="text-xs text-indigo-500"
                >
                  Start a new agent job
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
